summingbird-storm/src/main/scala/com/twitter/summingbird/storm/SpoutProvider.scala

```diff
@@ -0,0 +1,162 @@
+package com.twitter.summingbird.storm
+
+import backtype.storm.metric.api.IMetric
+import backtype.storm.task.TopologyContext
+import backtype.storm.topology.{ IRichSpout, TopologyBuilder }
+import com.twitter.algebird.Semigroup
+import com.twitter.algebird.util.summer.Incrementor
+import com.twitter.summingbird._
+import com.twitter.summingbird.batch.{ BatchID, Timestamp }
+import com.twitter.summingbird.online.{ Externalizer, executor }
+import com.twitter.summingbird.online.option.{ SourceParallelism, SummerBuilder }
+import com.twitter.summingbird.option.JobId
+import com.twitter.summingbird.planner.{ Dag, SummerNode }
+import com.twitter.summingbird.storm.planner.StormNode
+import com.twitter.tormenta.spout.Spout
+import com.twitter.summingbird.storm.Constants
+import com.twitter.summingbird.storm.option.SpoutStormMetrics
+import com.twitter.summingbird.storm.spout.KeyValueSpout
+
+case class SpoutProvider(storm: Storm, stormDag: Dag[Storm], node: StormNode, jobID: JobId)(implicit topologyBuilder: TopologyBuilder) {
+
+  private def getOrElse[T <: AnyRef: Manifest](stormDag: Dag[Storm], queryNode: StormNode = node, default: T): T = storm.getOrElse(stormDag, queryNode, default)
```

Comment:
Here and elsewhere, methods are taking `Dag[Storm]` and `StormNode` parameters that will never be anything other than just `stormDag` and `node`.  Let's just use those directly instead of taking the arguments?